packages/components/src/font-size-picker/index.tsx

```diff
@@ -169,6 +151,15 @@ const UnforwardedFontSizePicker = (
 		__( 'Currently selected font size: %s' ),
 		selectedOption.name
 	);
+
+	const [ valueNumber, valueUnit ] = parseNumberAndUnitFromSize( value );
+	const isValueUnitRelative =
+		!! valueUnit && [ 'em', 'rem' ].includes( valueUnit );
+	const [ , firstFontSizeUnit ] = parseNumberAndUnitFromSize(
+		fontSizes[ 0 ]?.size
+	);
+	const hasUnits = !! valueUnit || !! firstFontSizeUnit;
```

Comment:
Previously `hasUnits` was calculated by checking the type of `value` and the first option's `size`. If either was a string then we'd allow units. This isn't 100% correct though since technically a font size can be a string containing a number with no unit, i.e. `"30"`. This new logic uses `parseNumberAndUnitFromSize` to do things properly.

Reply:
I was wrong about this. See https://github.com/WordPress/gutenberg/pull/44598#issuecomment-1277002334.